ssl/quic/quic_impl.c

```diff
@@ -95,40 +122,26 @@ static int quic_raise_non_normal_error(QUIC_CONNECTION *qc,
     ERR_vset_error(ERR_LIB_SSL, reason, fmt, args);
     va_end(args);
 
-    if (qc != NULL)
-        qc->last_error = SSL_ERROR_SSL;
+    if (ctx != NULL) {
+        if (ctx->is_stream && ctx->xso != NULL)
+            ctx->xso->last_error = SSL_ERROR_SSL;
```

Comment:
The idea is that permanent errors are always connection level. But since the last_error state is essentially contextual to a specific thread - it contains state based on the last call that thread made to some stream object - it essentially needs to be treated as thread local. Hence this needs to be on the stream object. If we put it on the connection object it could get clobbered by other threads also making calls to stream objects on the same connection before the thread has a chance to call SSL_get_error.

Reply:
I assume this will need to be changed after #20857 (error handling design) is implemented.